### PR TITLE
Popover, Dropdown, PopoverEducational: support `forcePosition` prop

### PR DESCRIPTION
**Pull request by AlbertCarreras**
BREAKING CHANGE:

We are deprecating idealPosition: forceDown

```
yarn codemod detectManualReplacement ~/path/to/your/code
--component=Popover or Dropdown or PopoverEducational
--prop=idealDirection
--value=forceDown

```

Popover, Dropdown, PopoverEducational: support `forcePosition` prop

Test example here: https://deploy-preview-3744--gestalt.netlify.app/web/popover#Ideal-direction

### CODE REVIEW

File: packages/gestalt/src/Popover/InternalPopover.tsx [hunk 1]
```diff
@@ -18,7 +18,7 @@ type Props = {
   color?: Color;
   onKeyDown?: (arg1: { event: React.KeyboardEvent<HTMLElement> }) => void;
   id?: string;
-  idealDirection?: 'up' | 'right' | 'down' | 'left' | 'forceDown';
+  idealDirection?: 'up' | 'right' | 'down' | 'left' | 'forceDown' | 'forceRight';
```
Comment (elnanni) [inline packages/gestalt/src/Popover/InternalPopover.tsx:21]: Would this be a better approach if taken from packages/gestalt/src/utils/positioningTypes.ts using MainDirections?

**Comment by diyorbek** [comment]
@AlbertCarreras I think we should reconsider the API for force positioning. What if we'll need forceLeft, forceUp in the future? I would suggest to introduce `forceDirection: boolean` prop which will just disable auto-flipping/positioning behavior. We could just get rid of "force" prefixed positions.

**Comment by diyorbek** [comment]
The new prop doesn't have to be `forceDirection`. It could be `disableAutoFlip` or something like that. The point is this props will disable the auto-flipping beahvior of the popover.

File: packages/gestalt/src/Popover/Contents.tsx [hunk 2]
```diff
@@ -54,12 +66,18 @@ export default function Contents({
   overflow,
 }: Props) {
   const caretRef = useRef<HTMLElement | null>(null);
-  const idealPlacement = idealDirection ? DIRECTIONS_MAP[idealDirection] : 'top';
+
+  let idealPlacement: 'top' | 'right' | 'bottom' | 'left' = 'top';
```
Comment (diyorbek) [inline packages/gestalt/src/Popover/Contents.tsx:70]: Maybe this is more accurate: 

```suggestion
  let idealPlacement: Side = 'top';
```

File: packages/gestalt/src/Popover/usePopover.ts [hunk 3]
```diff
@@ -74,13 +71,20 @@ export default function usePopover({
...
   scrollBoundary,
   hideWhenReferenceHidden,
   onPositioned,
 }: Props): UseFloatingReturn {
-  const isForceDown = direction === 'forceDown';
-  const placement = isForceDown ? 'bottom' : direction;
+  let placement: Placement = direction ?? 'bottom';
+  const isRtl = typeof document === 'undefined' ? false : document?.dir === 'rtl';
+
+  if (isRtl && direction === 'left') {
```
Comment (diyorbek) [inline packages/gestalt/src/Popover/usePopover.ts:83]: @AlbertCarreras have you tested this bahvior? I though @floating-ui library already handled RTL. 🤔

**Review by diyorbek — APPROVED** [review]
LGTM

File: packages/gestalt/src/Popover/Contents.tsx (hunk 2 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Popover/Contents.tsx:70]: DONE

File: packages/gestalt/src/Popover/usePopover.ts (hunk 3 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/Popover/usePopover.ts:83]: It might flip accordingly... but if we force behavior idk... I'll test it anyways.

Comment (AlbertCarreras) [reply packages/gestalt/src/Popover/usePopover.ts:83]: just tested it... no, when forced... it stays i the right/left, cos they don't do end/start and we read LRT/LTR from the document

**Pull request merged**